pkg/obfuscate/sql_tokenizer.go

```diff
@@ -475,6 +479,18 @@ func (tkn *SQLTokenizer) Scan() (TokenKind, []byte) {
 				// want to cover for this use-case too (e.g. $1$some text$1$).
 				return tkn.scanPreparedStatement('$')
 			}
+
+			// A special case for a string starts with single $ but does not end with $.
+			// For example in SQLServer, you can have "MG..... OUTPUT $action, inserted.*"
+			// $action in the OUTPUT clause of a MERGE statement is a special identifier
+			// that returns one of three values for each row: 'INSERT', 'UPDATE', or 'DELETE'.
+			// See: https://docs.microsoft.com/en-us/sql/t-sql/statements/merge-transact-sql?view=sql-server-ver15
+			if tkn.cfg.DBMS == DBMSSQLServer && isLetter(tkn.lastChar) {
```

Comment:
We're testing for a letter, but the dollar sign could also be followed by a number. For example, `select $1` is also a valid statement - it will convert a number to the currency data type.

Reply:
Dollar sign followed by a number has been taken care as the first check, where we test if we see something like `$1` and run `scanPreparedStatement`. 
https://github.com/DataDog/datadog-agent/blob/87a921dbb9057b78a1eba640ee0bfff4a503263f/pkg/obfuscate/sql_tokenizer.go#L476-L481